src/app/store/test-records/selectors/test-records.selectors.ts

```diff
@@ -82,9 +83,12 @@ export const selectAmendedDefectData = createSelector(selectedAmendedTestResultS
 
 export const sectionTemplates = createSelector(testResultsFeatureState, state => state.sectionTemplates);
 
-export const isSameTestTypeId = createSelector(selectedAmendedTestResultState, selectedTestResultState, (testRecord, amendedTestRecord) => {
-  return testRecord?.testTypes[0].testTypeId === amendedTestRecord?.testTypes[0].testTypeId;
-});
+export const resultOfTestSelector = createSelector(toEditOrNotToEdit, testRecord => testRecord?.testTypes[0].testResult);
+
+export const isTestTypeKeySame = (key: keyof TestType) =>
+  createSelector(selectedAmendedTestResultState, selectedTestResultState, (testRecord, amendedTestRecord) => {
+    return testRecord?.testTypes[0][key] === amendedTestRecord?.testTypes[0][key];
```

Comment:
Is it guaranteed that `testRecord` and `amendedTestRecord` will always have at least one `testTypes`?

Reply:
They would both have at least one test type is they are defined